var/spack/repos/builtin/packages/kaldi/package.py

```diff
@@ -0,0 +1,90 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+from distutils.dir_util import copy_tree
+
+
+class Kaldi(Package):
+    """Kaldi is a toolkit for speech recognition written 
+    in C++ and licensed under the Apache License v2.0. 
+    Kaldi is intended for use by speech recognition researchers."""
+
+    homepage = "https://github.com/kaldi-asr/kaldi"
+    url      = "https://github.com/kaldi-asr/kaldi/archive/master.zip"
+
+    version('master', git='https://github.com/kaldi-asr/kaldi.git')
+
+    variant('shared', default=True,
+            description='build shared libraries')
+    variant('double', default=False,
+            description='build with double precision floats')
+    variant('cuda', default=False,
+            description='build with CUDA')
+
+    depends_on('blas')
+    depends_on('speex')
+    depends_on('openfst@1.6:')
+    depends_on('cuda', when='+cuda')
+    depends_on('sph2pipe', type='run')
+    depends_on('sctk', type='run')
+
+    def install(self, spec, prefix):
+        configure_args = [
+            '--shared', 
+            '--threaded-math',
+            '--speex-root=' + spec['speex'].prefix,
+            '--fst-root=' + spec['openfst'].prefix,
+            '--fst-version=' + str(spec['openfst'].version)
+        ]
+    
+        if '~shared' in spec:
+            configure_args.append('--static')
```

Comment:
So `+shared` builds shared and `~shared` builds shared and static. I'm not sure if that makes sense to me. I usually default to building shared and static. @davydden @citibeth Do you have any opinions on this?